lib/generators/hyrax/work_resource/templates/change_set.rb.erb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# Generated via
+#  `rails generate hyrax:work_resource <%= class_name %>`
+class <%= class_name %>ChangeSet < Hyrax::ChangeSet
```

Comment:
this should probably come with some extensive documentation about how change sets can be used.

if i'm understanding correctly, in a default application this won't be used at all, and editing this file will do nothing. is that right?

Reply:
You do not have to use this change set.  By default, it will behave exactly like Hyrax::ChangeSet.

I would imagine it would be fairly common for there to be custom validations.  That is how I am using it.

Use case:

We have two work types named Publication and Release.  A publication can have many release.  A release belongs to one and only one publication and cannot exist outside a publication.

One of my validations confirms that release.publication_id has a non-blank value.

Reply:
RE: Documentation - Do you envision links to external documentation for the validation options?

Reply:
i think links to external docs seem important.

but also, i think some example usage would help.

> I would imagine it would be fairly common for there to be custom validations

validations added here won't have any impact on any existing code, only on calls that initialize this specific ChangeSet or use the factory. i think it would be reasonable for a user seeing this generated into their codebase to wonder how its used in the engine. the answer "it isn't" will likely be surprising to folks, and i'd like to head that off.